src/main/editDelete/preProcess.ts

```diff
@@ -28,35 +28,66 @@ const injectAttributes: MapCallback<PartialWord, Word> = (word, index) => ({
   takeInfo: null,
 });
 
+const calculateBufferDurationBefore: (
+  word: Word,
+  prevWord: Word | null,
+  bufferBeforePadding: number
+) => number = (word, prevWord, bufferBeforePadding) => {
+  if (prevWord === null) {
+    return word.startTime;
+  }
+  const prevWordEndTime = prevWord.startTime + prevWord.duration;
+  const gapTime = word.startTime - prevWordEndTime;
+
+  // Half the gap used by this word, half by the previous word.
+  if (gapTime <= bufferBeforePadding) {
+    return roundToMs(gapTime);
+  }
+
+  return roundToMs(bufferBeforePadding);
+};
+
 const calculateBufferDurationAfter: (
   word: Word,
   nextWord: Word | null,
-  totalDuration: number
-) => number = (word, nextWord, totalDuration) => {
+  totalDuration: number,
+  bufferBeforePadding: number
+) => number = (word, nextWord, totalDuration, bufferBeforePadding) => {
   const wordEndTime = word.startTime + word.duration;
 
   if (nextWord === null) {
-    return totalDuration - wordEndTime;
+    return roundToMs(totalDuration - wordEndTime);
   }
   const gapTime = nextWord.startTime - wordEndTime;
 
-  return roundToMs(gapTime);
+  if (gapTime <= bufferBeforePadding) {
+    return roundToMs(0);
```

Comment:
There won't be floating point errors with this int literal, you can just `return 0;`